test/ViewExtensionLibraryTests/LibraryResourceProviderTests.cs

```diff
@@ -326,6 +327,52 @@ public void CustomNodeSearchElementLoadedType()
             Assert.AreEqual(IconUrl.DefaultPath, url.Path);
         }
 
+        [Test]
+        [Category("UnitTests")]
+        public void SearchNodeTest()
+        {
+            var nodeSearchModel = new NodeSearchModel();
+            var path = @"C:\temp\xyz.dyf";
+            var nodeName = "Code Block";
+            var expectedQualifiedName = $"dyf://123.456.somepackage.{nodeName}";
+
+            for (int i = 0; i < 100; i++)
+            {
+                nodeSearchModel.Add(
+                    new CustomNodeSearchElement(new Mock<ICustomNodeSource>().Object,
+                    new CustomNodeInfo(Guid.NewGuid(), $"Node-{i}", $"Node-{i}-Category{i}", $"Node-{i}-Description", path))
+                    );
+            }
+
+            nodeSearchModel.Add(
+                    new CustomNodeSearchElement(new Mock<ICustomNodeSource>().Object,
+                    new CustomNodeInfo(Guid.NewGuid(), nodeName, "123.456.somepackage", "Node-Description", path))
+                    );
+
+            var pathmanager = new Mock<IPathManager>();
+            var iconProvider = new IconResourceProvider(pathmanager.Object);
+
+            SearchResultDataProvider searchResultDataProvider = new SearchResultDataProvider(nodeSearchModel, iconProvider);
+
+            var extension = string.Empty;
+            var searchStream = searchResultDataProvider.GetResource("Code Block", out extension);
+
+            var data = GetLoadedTypesFromJson(searchStream);
+            List<LoadedTypeItem> types = data.loadedTypes;
+
+            Assert.AreEqual(types.Count, 1);
+            Assert.AreEqual(expectedQualifiedName, types[0].fullyQualifiedName);
```

Comment:
it looks good, but it seems you don't search for the custom node you added to the search model?

Reply:
updated